Update spectral match results visualizer

robinschmid opened the pull request:
Hey Tomas,

this PR is an update for the spectral matching results window. 
Ansgar has done a great job on the metadata panel. And I have changed the data model and the charts. Now every matching step is visualized in a different color:
 - filtered (black): Multiple options
    - 13C isotope filter (see image below)
    - crop to overlapping m/z range (first image, left side is black, because m/z range was different - or fragmentation energy was differnt)
 - aligned (green): Aligned signals in query and library spectrum (within m/z tolerance)
 - unaligned (orange)

![image](https://user-images.githubusercontent.com/10366914/58360574-cb6c4300-7e89-11e9-8cb8-7ffaab8f7613.png)
![image](https://user-images.githubusercontent.com/10366914/58360620-18e8b000-7e8a-11e9-950d-ccb12a4fb98e.png)

tomas-pluskal commented on src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/spectraidentification/spectraldatabase/SpectraIdentificationResultsWindow.java:
```diff
@@ -406,12 +430,15 @@ else if (smilesString != "N/A") {
+    constraintsMetaDataPanel.gridy = 3;
+    metaDataPanel.add(otherInfo, constraintsMetaDataPanel);
     SpectralIdentificationSpectralDatabaseTextPane investigator =
         new SpectralIdentificationSpectralDatabaseTextPane("Prinziple investigator: "
```
prinziple -> principal

tomas-pluskal commented on src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/mirrorspectra/MirrorScanWindow.java:
```diff
@@ -103,24 +112,96 @@ public void setScans(PeakListRow row, SpectralDBPeakIdentity db) {
+    // add data
+    DataPointsTag[] tags =
+        new DataPointsTag[] {DataPointsTag.ORIGINAL, DataPointsTag.FILTERED, DataPointsTag.ALIGNED};
+    Color[] colors = new Color[] {Color.black, new Color(0xF57C00), new Color(0x388E3C)};
```
This should definitely be defined as a public static final constant, with comments on what the hex colors are.

tomas-pluskal commented on src/main/java/net/sf/mzmine/util/maths/similarity/spectra/SpectraSimilarity.java:
```diff
@@ -18,16 +18,52 @@
+import net.sf.mzmine.util.SortingProperty;
+import net.sf.mzmine.util.scans.ScanAlignment;
+
 public class SpectraSimilarity {
```
Could you add a brief javadoc to explain the purpose of this class?

robinschmid commented:
I am going to finish this later today. I was away this weekend at the edersee lake.

robinschmid commented:
- Javadoc was added
- Moved SpectralSimilarity from util/maths/similarity/spectra  to util/scan/similarity
- Added an extra module to sort SpectralDBPeakIdentities of the search module
- correct spelling

robinschmid commented:
It is all ready for merging from my side.

Annexhc commented:
Can you explain why you have added a new module for the sorting?

robinschmid commented:
Modularity reasons, if someone searches multiple data bases, filters the results or changes the sorting manually - this module gives the options to ...
... sort
... filter

Annexhc commented:
Ah okay, good point that you may use different database for a single analysis. Good addition!

Annexhc commented:
From my side every is ready for merging as well.

tomas-pluskal commented:
Looks good!
